controllers/hvpa_controller.go

```diff
@@ -40,31 +42,40 @@ import (
 	"k8s.io/apimachinery/pkg/util/sets"
 	vpa_api "k8s.io/autoscaler/vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1beta2"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/controller"
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/handler"
-	"sigs.k8s.io/controller-runtime/pkg/manager"
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
 	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-var log = logf.Log.WithName("controller")
-
-// Add creates a new Hvpa Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
-// and Start it when the Manager is Started.
-func Add(mgr manager.Manager) error {
-	return add(mgr, newReconciler(mgr))
+// HvpaReconciler reconciles a Hvpa object
+type HvpaReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
 }
 
-// newReconciler returns a new reconcile.Reconciler
-func newReconciler(mgr manager.Manager) reconcile.Reconciler {
-	return &ReconcileHvpa{Client: mgr.GetClient(), scheme: mgr.GetScheme()}
-}
+var log = logf.Log.WithName("controller").WithName("hvpa")
 
 func updateEventFunc(e event.UpdateEvent) bool {
+	// If update event is for HVPA/HPA/VPA then we would want to reconcile unconditionally.
+	_, ok := e.ObjectOld.(*autoscalingv1alpha1.Hvpa)
```

Comment:
Might be better to use the [type switch case](https://tour.golang.org/methods/16) here.